CSharp/DevVmPowershell/Helpers/Implementations/AgentHelper.cs

```diff
@@ -86,66 +86,105 @@ private async Task<List<int>> GetAgentArtifactIdsAsync(string agentName)
-				Console.WriteLine($"Total Available Agent Servers for Agent Type: {agentServerResponseList.Count}");
-				return agentServerResponseList;
+				agentServerResponseList.Add(new AgentServerResponseModel()
+				{
+					Type = obj["Type"].ToString(),
+					ProcessorCores = Convert.ToInt32(obj["ProcessorCores"].ToString()),
+					NumberOfAgents = Convert.ToInt32(obj["NumberOfAgents"].ToString()),
+					ArtifactID = Convert.ToInt32(obj["ArtifactID"].ToString()),
+					Name = obj["Name"].ToString()
+				});
 			}
+
+			return agentServerResponseList;
 		}
 
 		private async Task CreateAgentAsync(int agentTypeArtifactId, int agentServerArtifactId, decimal defaultInterval, int defaultLoggingLevel)
 		{
-			AgentRequest newAgentRequest = new AgentRequest
+			var agentCreateRequestModel = new
 			{
-				Enabled = Constants.Agents.ENABLE_AGENT,
-				Interval = defaultInterval,
-				Keywords = Constants.Agents.KEYWORDS,
-				Notes = Constants.Agents.NOTES,
-				LoggingLevel = defaultLoggingLevel,
-				AgentType = new Securable<ObjectIdentifier>(
-					new ObjectIdentifier
+				AgentRequest = new
+				{
+					AgentType = new
 					{
-						ArtifactID = agentTypeArtifactId
-					}),
-				AgentServer = new Securable<ObjectIdentifier>(
-					new ObjectIdentifier
+						Value = new
+						{
+							ArtifactID = agentTypeArtifactId
+						}
+					},
+					AgentServer = new
 					{
-						ArtifactID = agentServerArtifactId
-					})
+						Value = new
+						{
+							ArtifactID = agentServerArtifactId
+						}
+					},
+					Enabled = Constants.Agents.ENABLE_AGENT,
+					Interval = defaultInterval,
+					Keywords = Constants.Agents.KEYWORDS,
+					Notes = Constants.Agents.NOTES,
+					LoggingLevel = defaultLoggingLevel
+				}
 			};
 
-			using (Relativity.Services.Interfaces.Agent.IAgentManager agentManager = ServiceFactory.CreateProxy<Relativity.Services.Interfaces.Agent.IAgentManager>())
+			string createPayload = JsonConvert.SerializeObject(agentCreateRequestModel);
+			HttpClient httpClient = RestHelper.GetHttpClient(InstanceAddress, AdminUsername, AdminPassword);
+			HttpResponseMessage response = await RestHelper.MakePostAsync(httpClient, Constants.Connection.RestUrlEndpoints.Agent.EndpointUrl, createPayload);
+			if (!response.IsSuccessStatusCode)
 			{
-				int newAgentArtifactId = await agentManager.CreateAsync(Constants.EDDS_WORKSPACE_ARTIFACT_ID, newAgentRequest);
-				Console.WriteLine($"{nameof(newAgentArtifactId)}: {newAgentArtifactId}");
+				throw new Exception("Failed to Create Agent");
 			}
 		}
 
 		private async Task DeleteAgentAsync(int agentArtifactId)
 		{
-			using (Relativity.Services.Interfaces.Agent.IAgentManager agentManager = ServiceFactory.CreateProxy<Relativity.Services.Interfaces.Agent.IAgentManager>())
+			string url = $"Relativity.rest/api/relativity.agents/workspace/-1/agents/{agentArtifactId}";

```

Comment:
- [ ]  Move the url to the Constants class

Reply:
Done